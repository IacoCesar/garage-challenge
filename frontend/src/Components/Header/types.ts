export interface IHeader {
  testID?: string
  isHomeScreen?: boolean
  title?: string
  callHandleBack?: (navigation: any) => void
  callHandleChangeVisualMode?: (themeSelector: string) => void
}

export interface IProps extends IHeader {
  themeSelector: string
}
