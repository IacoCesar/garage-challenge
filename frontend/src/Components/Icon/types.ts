export interface IProps {
  testID?: string
  size: number
  name: string
  color: string
  type?: string
}
