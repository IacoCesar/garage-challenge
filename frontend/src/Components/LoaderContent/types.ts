export interface IProps {
  testID?: string
  size: number
  color: string
}
