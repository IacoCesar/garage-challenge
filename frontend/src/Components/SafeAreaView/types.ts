export interface IProps {
  testID?: string
  children: JSX.Element
}

export interface IThemeProps {
  themeSelector: string
}
