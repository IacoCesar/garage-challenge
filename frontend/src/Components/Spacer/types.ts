export interface IProps {
  testID?: string
  min: number
  max: number
}
