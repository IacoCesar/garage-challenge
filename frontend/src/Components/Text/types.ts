export interface ITextProps {
  color?: string
  fontSize?: number
  isBold?: boolean
}

export interface IProps extends ITextProps {
  label: string
  testID?: string
}
