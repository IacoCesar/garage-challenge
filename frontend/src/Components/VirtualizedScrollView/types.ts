export interface IProps {
  testID?: string
  themeSelector: string
  children?: JSX.Element
}
