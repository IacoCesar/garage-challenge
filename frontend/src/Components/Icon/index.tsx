import React from 'react'
import { MaterialIcons, AntDesign } from '@expo/vector-icons'
import { IProps } from './types'
import { assignTestId } from '@/Helpers/TestId'

const Component: React.FC<IProps> = ({
  testID = 'Icon',
  size,
  name,
  color,
  type
}) => {
  return type === 'material' ? (
    <MaterialIcons
      {...assignTestId('View', testID)}
      size={size}
      name={name}
      color={color}
    />
  ) : (
    <AntDesign
      {...assignTestId('View', testID)}
      size={size}
      name={name}
      color={color}
    />
  )
}

export default Component
