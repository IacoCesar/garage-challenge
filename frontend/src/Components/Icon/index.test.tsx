import React from 'react'
import { IProps } from './types'
import { mockTestId } from '@/Helpers/TestId'
import Icon from '.'
import { render } from '@/Helpers/TestHelpers'

describe('Icon', () => {
  let props: IProps = {
    testID: 'Icon',
    size: 23,
    name: 'star',
    color: 'white',
    type: 'material'
  }

  const iconElement = mockTestId('View', `${props.testID!}`).testID

  it(`should render ${iconElement}`, async () => {
    const wrapper = render(<Icon {...props} />)

    const el_container = wrapper.getByTestId(iconElement)
    expect(el_container).toBeTruthy()
  })

  it('should render snapshot', async () => {
    const wrapper = render(<Icon {...props} />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
