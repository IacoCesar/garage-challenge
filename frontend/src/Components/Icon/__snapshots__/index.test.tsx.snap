// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render snapshot 1`] = `
<View
  accessible={false}
  color="white"
  name="star"
  size={23}
  testID="View_Icon"
/>
`;
